tqec/detectors/measurement_map.py

```diff
@@ -1,23 +1,33 @@
 import cirq
 
 
+def flatten(obj: cirq.Moment | cirq.AbstractCircuit) -> cirq.Circuit:
+    # From https://github.com/quantumlib/Cirq/issues/3783
+    if isinstance(obj, cirq.Moment):
+        flattenable = [op for op in obj if isinstance(op, cirq.CircuitOperation)]
+        if not flattenable:
+            return cirq.Circuit(obj)
+        return cirq.Circuit.zip(
+            cirq.Circuit(obj - flattenable),
+            *[flatten(e.mapped_circuit(True)) for e in flattenable],
+        )
+    return cirq.Circuit([moment for e in obj for moment in flatten(e)])
```

Comment:
Ah, this is a recursive function! I still think separating them out makes this a little more comprehensible.